node/src/main/scala/coop/rchain/node/effects/TcpTransportLayer.scala

```diff
@@ -1,27 +1,65 @@
 package coop.rchain.node.effects
 
-import scala.concurrent.duration._
+import java.io.File
+
 import coop.rchain.comm._, CommError._
 import coop.rchain.comm.protocol.routing._
 import coop.rchain.p2p.effects._
 import coop.rchain.metrics.Metrics
-import io.grpc.{ManagedChannel, ManagedChannelBuilder, Server, ServerBuilder}
 
 import cats._, cats.data._, cats.implicits._
 import coop.rchain.catscontrib._, Catscontrib._, ski._, TaskContrib._
 
-import scala.concurrent.duration.{Duration, MILLISECONDS}
+import scala.concurrent.duration._
 import scala.util.Try
-import scala.concurrent.{Await, Promise}
+import scala.concurrent.Await
 import scala.concurrent.{ExecutionContext, Future}
+import io.grpc.netty._
+import io.netty.handler.ssl.{ClientAuth, SslContext}
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory
 
-class TcpTransportLayer[F[_]: Monad: Capture: Metrics: Futurable](host: String, port: Int)(
-    src: PeerNode)(implicit executionContext: ExecutionContext)
+class TcpTransportLayer[F[_]: Monad: Capture: Metrics: Futurable](
```

Comment:
So we keep the `Await` eventually ?

Reply:
yes, because GRPC knows only about `deadlines` and not `timeouts`. Deadlines are fixed points in time. This would work for now. But at some point we want to keep the channels open in a `State` Monad.